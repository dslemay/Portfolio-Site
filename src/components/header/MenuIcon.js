import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { css } from 'emotion';
import themeUtils from '../themeUtils';

const MenuContainer = styled('div')`
  position: relative;
  right: 25px;
  top: 0;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Icon = styled('div')`
  position: relative;
  height: 20px;
  width: 30px;
  cursor: pointer;
  z-index: 10;
`;

const MenuLine = css`
  background-color: ${themeUtils.lightAccent};
  height: 3px;
  width: 100%;
  border-radius: 2px;
  position: absolute;
  left: 0;
  transition: all 0.25s cubic-bezier(0.175, 0.885, 0.32, 1.275);
`;

const Line1 = styled('span')`
  ${MenuLine};
  top: 0;
`;

const Line1Open = css`
  transform: translateY(10px) translateY(-50%) rotate(-135deg);
`;

const Line2 = styled('span')`
  ${MenuLine};
  top: 0;
  bottom: 0;
  margin: auto;
`;

const Line2Open = css`
  opacity: 0;
`;

const Line3 = styled('span')`
  ${MenuLine};
  bottom: 0;
`;

const Line3Open = css`
  transform: translateY(-10px) translateY(50%) rotate(135deg);
`;

const MenuIcon = ({ isOpen, toggleOpen }) => (
  <MenuContainer onClick={() => toggleOpen()}>
    <Icon>
      <Line1 className={isOpen ? Line1Open : ''} />
      <Line2 className={isOpen ? Line2Open : ''} />
      <Line3 className={isOpen ? Line3Open : ''} />
    </Icon>
  </MenuContainer>
);

MenuIcon.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggleOpen: PropTypes.func.isRequired,
};

export default MenuIcon;
