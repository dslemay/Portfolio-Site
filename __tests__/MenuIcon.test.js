import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MenuIcon from '../src/components/header/MenuIcon';

describe('MenuIcon', () => {
  let props;
  let mountedMenuIcon;

  const menuIcon = () => {
    if (!mountedMenuIcon) {
      mountedMenuIcon = mount(<MenuIcon {...props} />);
    }
    return mountedMenuIcon;
  };

  beforeEach(() => {
    props = {
      toggleOpen: jest.fn(),
      isOpen: false,
    };

    mountedMenuIcon = undefined;
  });

  it('calls toggleOpen fn when clicking the div containing the icon', () => {
    menuIcon()
      .find('div')
      .first()
      .simulate('click');
    expect(menuIcon().props().toggleOpen).toBeCalled();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MenuIcon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
