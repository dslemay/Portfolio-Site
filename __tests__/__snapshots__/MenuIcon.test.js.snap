// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcon renders correctly 1`] = `
<div
  className="css-5eaigi css-183znxx0"
  onClick={[Function]}
>
  <div
    className="css-m56tt4 css-183znxx1"
  >
    <span
      className="css-190cys6 css-183znxx2"
    />
    <span
      className="css-1nmu6z9 css-183znxx3"
    />
    <span
      className="css-g78e9d css-183znxx4"
    />
  </div>
</div>
`;
